backend/src/middleware/filterBubble/filterBubble.spec.js

```diff
@@ -52,33 +76,44 @@ afterEach(async () => {
 
 describe('Filter posts by author is followed by sb.', () => {
   describe('given an authenticated user', () => {
-    let authenticatedClient
-
     beforeEach(async () => {
-      const headers = await login(currentUserParams)
-      authenticatedClient = new GraphQLClient(host, { headers })
+      authenticatedUser = await user.toJson()
     })
 
     describe('no filter bubble', () => {
       it('returns all posts', async () => {
-        const query = '{ Post(filter: { }) { title } }'
+        const postQuery = gql`
+          {
+            Post(filter: {}) {
+              title
+            }
+          }
+        `
         const expected = {
-          Post: [
-            { title: 'This is some random post' },
-            { title: 'This is the post of a followed user' },
-          ],
+          data: {
+            Post: [
+              { title: 'This is some random post' },
+              { title: 'This is the post of a followed user' },
+            ],
+          },
         }
-        await expect(authenticatedClient.request(query)).resolves.toEqual(expected)
+        await expect(query({ query: postQuery })).resolves.toMatchObject(expected)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T09:19:55Z" title="Tuesday, September 3rd 2019, 11:19:55 am +02:00">Sep 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -52,33 +76,44 @@ afterEach(async () => {
 
 describe('Filter posts by author is followed by sb.', () => {
   describe('given an authenticated user', () => {
-    let authenticatedClient
-
     beforeEach(async () => {
-      const headers = await login(currentUserParams)
-      authenticatedClient = new GraphQLClient(host, { headers })
+      authenticatedUser = await user.toJson()
     })
 
     describe('no filter bubble', () => {
       it('returns all posts', async () => {
-        const query = '{ Post(filter: { }) { title } }'
+        const postQuery = gql`
+          {
+            Post(filter: {}) {
+              title
+            }
+          }
+        `
         const expected = {
-          Post: [
-            { title: 'This is some random post' },
-            { title: 'This is the post of a followed user' },
-          ],
+          data: {
+            Post: [
+              { title: 'This is some random post' },
+              { title: 'This is the post of a followed user' },
+            ],
+          },
         }
-        await expect(authenticatedClient.request(query)).resolves.toEqual(expected)
+        await expect(query({ query: postQuery })).resolves.toMatchObject(expected)
```


You changed this `toMatchObject` because there are other results of `query`, which you are not interested in testing?
`Use .toMatchObject to check that a JavaScript object matches a subset of the properties of an object`
https://jestjs.io/docs/en/expect#tomatchobjectobject

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-03T12:35:28Z" title="Tuesday, September 3rd 2019, 2:35:28 pm +02:00">Sep 3, 2019</time>_

---


Yes, I found out that `toMatchObject` is the perfect matcher to check a graphql response. You can even check the `errors` array that way. It is concise (you expect only the minimum) yet robust (no false negatives).

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T13:54:31Z" title="Tuesday, September 3rd 2019, 3:54:31 pm +02:00">Sep 3, 2019</time>_

---


I think I will try to use it instead of `expect.objectContaining`